samples/mentions.html

```diff
@@ -0,0 +1,257 @@
+				{ id: 1, avatar: 'm_1', fullname: 'Charles Flores', username: 'cflores' },
+				{ id: 2, avatar: 'm_2', fullname: 'Gerald Jackson', username: 'gjackson' },
+				{ id: 3, avatar: 'm_3', fullname: 'Wayne Reed', username: 'wreed' },
+				{ id: 4, avatar: 'm_4', fullname: 'Louis Garcia', username: 'lgarcia' },
+				{ id: 5, avatar: 'm_5', fullname: 'Roy Wilson', username: 'rwilson' },
+				{ id: 6, avatar: 'm_6', fullname: 'Matthew Nelson', username: 'mnelson' },
+				{ id: 7, avatar: 'm_7', fullname: 'Randy Williams', username: 'rwilliams' },
+				{ id: 8, avatar: 'm_8', fullname: 'Albert Johnson', username: 'ajohnson' },
+				{ id: 9, avatar: 'm_9', fullname: 'Steve Roberts', username: 'sroberts' },
+				{ id: 10, avatar: 'm_10', fullname: 'Kevin Evans', username: 'kevans' },
+
+				{ id: 11, avatar: 'w_1', fullname: 'Mildred Wilson', username: 'mwilson' },
+				{ id: 12, avatar: 'w_2', fullname: 'Melissa Nelson', username: 'mnelson' },
+				{ id: 13, avatar: 'w_3', fullname: 'Kathleen Allen', username: 'kallen' },
+				{ id: 14, avatar: 'w_4', fullname: 'Mary Young', username: 'myoung' },
+				{ id: 15, avatar: 'w_5', fullname: 'Ashley Rogers', username: 'arogers' },
+				{ id: 16, avatar: 'w_6', fullname: 'Debra Griffin', username: 'dgriffin' },
+				{ id: 17, avatar: 'w_7', fullname: 'Denise Williams', username: 'dwilliams' },
+				{ id: 18, avatar: 'w_8', fullname: 'Amy James', username: 'ajames' },
+				{ id: 19, avatar: 'w_9', fullname: 'Ruby Anderson', username: 'randerson' },
+				{ id: 20, avatar: 'w_10', fullname: 'Wanda Lee', username: 'wlee' }
+			];
+
+		CKEDITOR.replace( 'editor1', {
+			plugins: 'mentions,emoji,basicstyles,undo,link,wysiwygarea,toolbar',
+			height: 150,
+			toolbar: [
+				{ name: 'document', items: [ 'Undo', 'Redo' ] },
+				{ name: 'basicstyles', items: [ 'Bold', 'Italic', 'Strike' ] },
+				{ name: 'links', items: [ 'Link', 'Unlink' ] }
+			],
+			mentions: [
+				{
+					feed: dataFeed,
+					itemTemplate:
+					'<li data-id="{id}">' +
+						'<img class="photo" src="assets/mentions/img/{avatar}.jpg" />' +
+						'<strong class="username">{username}</strong>' +
+						'<span class="fullname">{fullname}</span>' +
+					'</li>',
+					outputTemplate: '<a href="mailto:{username}@example.com">@{username}</a><span>&nbsp;</span>',
+					minChars: 0
+				}
+			],
+			on: {
+				instanceReady: function( evt ) {
+					document.getElementById( 'send' ).addEventListener( 'click', function() {
+						evt.editor.showNotification( 'Your message has been send! ' +
+						'However this is only example chat and we cannot make your post public.',
+						'success', 8000 );
+					} );
+				}
+			}
+		} );
+
+		function dataFeed( opts, callback ) {
+			var data = users.filter( function( item ) {
+				return item.username.indexOf( opts.query.toLowerCase() ) == 0;
+			} );
+
+			data = data.sort( function( a, b ) {
+				var nameA = a.fullname.toUpperCase(),
+					nameB = b.fullname.toUpperCase();
+
+				if (nameA < nameB) {
```

Comment:
This function could be much, much simplified. First, don't be afraid to use `if`, `else if` `else` construction if there's a need to.

However in this case you could just reuse `localeCompare()` function (I'm sure you used it at some point 😉).

Reply:
I'd also extract name of the property used for filtering, ordering into a variable, so that would result with _something_ like that:

```js
function dataFeed( opts, callback ) {
	var filteredProperty = 'username';,
		data = users.filter( function( item ) {
			return item[ filteredProperty ].indexOf( opts.query.toLowerCase() ) == 0;
		} );

	data = data.sort( function( a, b ) {
		return a[ matchProperty ].localeCompare( b[ matchProperty ], undefined, { sensitivity: 'accent' } );
	} );

	callback( data );
}
```